common/types/mapper/proto/api.go

```diff
@@ -1766,18 +1766,52 @@ func FromGetTaskListsByDomainResponse(t *types.GetTaskListsByDomainResponse) *ap
 	if t == nil {
 		return nil
 	}
-	return &apiv1.GetTaskListsByDomainResponse{
-		TaskListNames: t.GetTaskListNames(),
+	resp := &apiv1.GetTaskListsByDomainResponse{}
+	decisionTaskList := t.GetDecisionTaskListMap()
+	activityTaskList := t.GetActivityTaskListMap()
+	var decisionTaskListMap, activityTaskListMap map[string]*apiv1.DescribeTaskListResponse
+	if len(decisionTaskList) > 0 {
+		decisionTaskListMap = make(map[string]*apiv1.DescribeTaskListResponse, len(decisionTaskList))
+		for name, tl := range decisionTaskList {
+			decisionTaskListMap[name] = FromDescribeTaskListResponse(tl)
+		}
+	}
```

Comment:
nit: can we create another set of From/To methods for the response map and reuse it for both activity and decision task list map? Then this method can also be simplified to something like
```
if t == nil {
    return nil
}

return &apiv1.GetTaskListsByDomainResponse{
    DecisionTaskListMap: FromDescribeTaskListResponseMap(t.GetDecisionTaskListMap()),
    ActivityTaskListMap: FromDescribeTaskListResponseMap(t.GetActivityTaskListMap()),
}
```

Reply:
good point. Update it